### PR TITLE
#60 [Final] Update User DB Password Patch API 

### PR DESCRIPTION
**Pull request by AliceHuang1027**
> server.js 
add app.patch(...)
>server.test.js
add patch:jest.fn() to each test
>userRoutes.js
add updateDBPassword function to routes object
>userRoutes.test.js
add updateDBPassword function tests

server.js 33-34 has error:
'Jest: "global" coverage threshold for statements (100%) not met: 98.98%
Jest: "global" coverage threshold for lines (100%) not met: 98.97%
Jest: "global" coverage threshold for functions (100%) not met: 96.67%'

Screen Shot 2020-06-10 at 11 47 22 PM

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -29,9 +29,15 @@ const stopServer = () => {
...
     logger.info("DB has been closed")
+<<<<<<< HEAD
+    server.close(() => { 
+      logger.info("The server has been closed")
+      resolve() 
+=======
     server.close(() => {
       logger.info("The server has been closed")
       resolve()
+>>>>>>> fc12258a76a170c081cdad3d3a83bf4608aa4f49
```
Comment (kimong1) [inline src/server.js:40]: This part is duplicated

File: src/server.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ const startServer = async (portNumber) => {
     app.use(express.json())
 
     app.post('/api/notifications', resetPassword)
-
+    app.patch('/api/users/:id',updateDBPassword)
```
Comment (kimong1) [inline src/server.js:20]: This needs to be tested too. Doesn't it?

Comment (AliceHuang1027) [reply src/server.js:20]: yea i changed my original test to meet carlos' code. i was thinking his code in the server.js fire was right...😕

File: src/routes/userRoutes.js [hunk 3]
```diff
@@ -35,5 +35,33 @@ routes.resetPassword = async (req, res) => {
...
+      if(!userAccount){
+        logger.info(`account does not exist`)
+        res.status(400).json({error:{message:"account does not exist"}})
+        return
+    }
+  try {
+
+    await setDBPassword(userAccount,req.body.password) 
+    logger.info(`user ${userAccount.id} updates password`)
+    return res.status(200).json('success')
```
Comment (greenc123) [inline src/routes/userRoutes.js:58]: Return an object to be consistent with your other json calls

```
return res.status(200).json({message: "success"})

Comment (AliceHuang1027) [reply src/routes/userRoutes.js:58]: i guess just success is fine as long as it's json. plz check the exercise in js5 for reference:)

**Pull request closed without merging**